dags/elsevier/metadata_parser.py

```diff
@@ -0,0 +1,119 @@
+import os
+from datetime import datetime
+from common.parsing.parser import IParser
+
+from common.parsing.xml_extractors import CustomExtractor
+from common.utils import extract_text, get_license_type_and_version_from_url
+from structlog import get_logger
+
+
+class ElsevierMetadataParser(IParser):
+    def __init__(self, doi, file_path) -> None:
+        self.dois = doi
+        self.file_path = file_path
+        self.year = None
+        self.journal_doctype = None
+        self.collaborations = []
+        self.logger = get_logger().bind(class_name=type(self).__name__)
+        self.extractors = [
+            CustomExtractor(
+                destination="journal_title",
+                extraction_function=self._get_journal_title,
+                required=True,
+            ),
+            # CustomExtractor(
+            #     destination="published_date",
+            #     extraction_function=self._get_published_date,
+            #     required=True,
+            # ),
+            CustomExtractor(
+                destination="date_published",
+                extraction_function=self._get_published_date,
+                # required=True,
+            ),
+            CustomExtractor(
+                destination="journal_year",
+                extraction_function=self._get_journal_year,
+                required=True,
+            ),
+            CustomExtractor(
+                destination="collections",
+                extraction_function=self._get_collections,
+                # required=True,
+            ),
+            CustomExtractor(
+                destination="license",
+                extraction_function=self._get_license,
+                # required=True,
+            ),
+            # CustomExtractor(
+            #     destination="local_files",
```

Comment:
should we keep it? It might be useful in the future. This field is in the current hepcrawl, but it doesn't pass the validation since it is not in the [schema](https://repo.qa.scoap3.org/schemas/hep.json). 
In schema we have _files:
```
 "_files": {
            "title": "File schema",
            "type": "array",
            "properties": {
                "key": {
                    "minLength": 1,
                    "type": "string"
                },
                "checksum": {
                    "minLength": 1,
                    "type": "string"
                },
                "size": {
                    "type": "integer"
                },
                "bucket": {
                    "minLength": 1,
                    "type": "string"
                },
                "version_id": {
                    "minLength": 1,
                    "type": "string"
                },
                "filetype": {
                    "minLength": 1,
                    "type": "string"
                }
            }
        },
```